### PR TITLE
fixed logging part of telemetry

### PR DESCRIPTION
**Pull request by tamilmani1989**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
This PR addresses 
1. prevents multi process access same telemetry lofile. only telemetry server can access log file.
2. included more meaningful logs

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: cni/network/plugin/main.go [hunk 1]
```diff
@@ -133,6 +133,52 @@ func handleIfCniUpdate(update func(*skel.CmdArgs) error) (bool, error) {
...
+
+	if err := common.StartProcess(path); err != nil {
+		log.Printf("[Telemetry] Failed to start telemetry service process :%v", err)
+		return err
+	}
+
+	log.Printf("[cni] Telemetry service started")
+
+	for attempt := 0; attempt < 5; attempt++ {
+		if telemetry.CheckIfSockExists() {
```
Comment (jaer-tsun) [inline cni/network/plugin/main.go]: telemetry.SockExists() ?

Why are there 5 attempts?

File: cni/network/plugin/main.go [hunk 2]
```diff
@@ -133,6 +133,52 @@ func handleIfCniUpdate(update func(*skel.CmdArgs) error) (bool, error) {
...
+		time.Sleep(200 * time.Millisecond)
+	}
+
+	return nil
+}
+
+func connectToTelemetryService(tb *telemetry.TelemetryBuffer) {
+	path := fmt.Sprintf("%v/%v", telemetry.CniInstallDir, telemetry.TelemetryServiceProcessName)
+
+	for attempt := 0; attempt < 2; attempt++ {
```
Comment (jaer-tsun) [inline cni/network/plugin/main.go]: why are there 2 attempts?

File: cni/network/plugin/main.go [hunk 3]
```diff
@@ -133,6 +133,52 @@ func handleIfCniUpdate(update func(*skel.CmdArgs) error) (bool, error) {
...
+		if err := tb.Connect(); err != nil {
+			log.Printf("Connection to telemetry socket failed: %v", err)
+			tb.Cleanup(telemetry.FdName)
+
+			if isExists, _ := common.CheckIfFileExists(path); !isExists {
```
Comment (jaer-tsun) [inline cni/network/plugin/main.go]: FileExists()?

File: telemetry/telemetry.go [hunk 4]
```diff
@@ -440,9 +430,10 @@ func (reportMgr *ReportManager) ReportToBytes() (report []byte, err error) {
 		err = fmt.Errorf("[Telemetry] Invalid report type")
 	}
 
-	if err == nil {
```
Comment (jaer-tsun) [inline telemetry/telemetry.go]: why did we change this? It's the same behaviour right?

File: telemetry/telemetry_test.go [hunk 5]
```diff
@@ -214,6 +226,10 @@ func TestClientCloseTelemetryConnection(t *testing.T) {
 		go tb.BufferAndPushData(0)
 	}
 
+	if !CheckIfSockExists() {
```
Comment (jaer-tsun) [inline telemetry/telemetry_test.go]: SockExists()

File: cni/network/plugin/main.go (hunk 1 above)
Comment (tamilmani1989) [reply cni/network/plugin/main.go]: just a calculated number. wait for second and then exit if telemetry socket doesn;t exist.

File: cni/network/plugin/main.go (hunk 2 above)
Comment (tamilmani1989) [reply cni/network/plugin/main.go]: same here..retry for 2 attempts and give up

File: cni/network/plugin/main.go (hunk 3 above)
Comment (tamilmani1989) [reply cni/network/plugin/main.go]: check if telemetryservice.exe exists and then start process..vivek suggested to have this logic..

File: telemetry/telemetry_test.go (hunk 5 above)
Comment (tamilmani1989) [reply telemetry/telemetry_test.go]: ok

File: telemetry/telemetry.go (hunk 4 above)
Comment (tamilmani1989) [reply telemetry/telemetry.go]: i checked this and it didn't work
> err= fmt.Errorf("[Telemetry] Invalid report type")

outside the switch case its showing nil

**Pull request merged**